momento-sdk/src/main/java/momento/sdk/config/transport/GrpcConfiguration.java

```diff
@@ -3,33 +3,61 @@
 import static momento.sdk.ValidationUtils.ensureRequestDeadlineValid;
 
 import java.time.Duration;
+import java.util.OptionalInt;
 import javax.annotation.Nonnull;
+import momento.sdk.internal.GrpcChannelOptions;
 
 /** Abstracts away the gRPC configuration tunables. */
 public class GrpcConfiguration {
 
   private final Duration deadline;
   private final int minNumGrpcChannels;
+  private final OptionalInt maxMessageSize;
```

Comment:
interesting, i'd never seen `OptionalInt` before, only `Optional<Int>`. but this seems fine.

is there a reason not to make the boolean one Optional as well?

Reply:
I was only able to get `OptionalInt` to compile (not `Optional<int>`) and there was no OptionalBoolean equivalent.

Did another quick search and found [this](https://schneide.blog/2019/04/23/javas-optionalint-et-al-versus-optionalt/) so I guess I just missed the fact that Java has both `int` and `Integer`. Will switch to using `Optional<Integer>` and `Optional<Boolean>` then.

Reply:
yeah, generics only work with the "boxed" data types, sorry I forgot to mention that.